### PR TITLE
Standardize inputs and onchanges

### PR DESCRIPTION
**Pull request by zarsky-broad**
Standardize input components so they're all called with `h`, and all return their new value directly to their `onChange`.

### CODE REVIEW

**Review by breilly2 — APPROVED** [review]
👍 Nice! I really like the simplification of `onChange` functions.

**Review by elizzhao — APPROVED** [review]

File: src/components/input.js [hunk 1]
```diff
@@ -289,9 +291,7 @@ export class AutocompleteTextInput extends Component {
         ])
       },
       renderSuggestion: v => v,
-      renderInputComponent: inputProps => {
-        return textInput({ ...props, ...inputProps, style, type: 'search' })
-      },
+      renderInputComponent: inputProps => h(TextInput, { ...props, ...inputProps, style, type: 'search' }),
```
Comment (panentheos) [inline src/components/input.js]: This should also use `nativeOnChange`, right now the workflow config is crashing.

File: src/components/input.js [hunk 2]
```diff
@@ -359,6 +358,6 @@ export const TextArea = ({ onChange, ...props }) => {
     as: 'textarea',
     className: 'focus-style',
     style: styles.textarea,
-    onChange: onChange ? (e => onChange(e.target.value)) : undefined
+    onChange: e => onChange(e.target.value)
```
Comment (panentheos) [inline src/components/input.js]: nit: React will complain in the console if you try to make a `readOnly` component with an `onChange` handler, which is why it was written this way. I don't think we actually do this anywhere currently, but it's good to be well-behaved for this case anyway.

**Review by panentheos — APPROVED** [review]
Looks good, but see comment.

**Pull request merged**